templates/adding_a_new_model/open_model_proposals/ADD_GLM.md

```diff
@@ -0,0 +1,1129 @@
+
+4.  [ ] Created script that successfully runs forward pass using
+    original repository and checkpoint
+
+5.  [ ] Successfully opened a PR and added the model skeleton to Transformers
+
+6.  [ ] Successfully converted original checkpoint to Transformers
+    checkpoint
+
+7.  [ ] Successfully ran forward pass in Transformers that gives
+    identical output to original checkpoint
+
+8.  [ ] Finished model tests in Transformers
+
+9.  [ ] Successfully added Tokenizer in Transformers
+
+10. [ ] Run end-to-end integration tests
+
+11. [ ] Finished docs
+
+12. [ ] Uploaded model weights to the hub
+
+13. [ ] Submitted the pull request for review
+
+14. [ ] (Optional) Added a demo notebook
+
+To begin with, we usually recommend to start by getting a good
+theoretical understanding of `GLM`. However, if you prefer to
+understand the theoretical aspects of the model *on-the-job*, then it is
+totally fine to directly dive into the `GLM`'s code-base. This
+option might suit you better, if your engineering skills are better than
+your theoretical skill, if you have trouble understanding
+`GLM`'s paper, or if you just enjoy programming much more than
+reading scientific papers.
+
+### 1. (Optional) Theoretical aspects of GLM
+
+You should take some time to read *GLM's* paper, if such
+descriptive work exists. There might be large sections of the paper that
+are difficult to understand. If this is the case, this is fine - don't
+worry! The goal is not to get a deep theoretical understanding of the
+paper, but to extract the necessary information required to effectively
+re-implement the model in 🤗 Transformers. That being said, you don't
+have to spend too much time on the theoretical aspects, but rather focus
+on the practical ones, namely:
+
+-   What type of model is *GLM*? BERT-like encoder-only
+    model? GPT2-like decoder-only model? BART-like encoder-decoder
+    model? Look at the `model_summary` if
+    you're not familiar with the differences between those.
+-   What are the applications of *GLM*? Text
+    classification? Text generation? Seq2Seq tasks, *e.g.,*
+    summarization?
+-   What is the novel feature of the model making it different from
+    BERT/GPT-2/BART?
+-   Which of the already existing [🤗 Transformers
+    models](https://huggingface.co/transformers/#contents) is most
+    similar to *GLM*?
+-   What type of tokenizer is used? A sentencepiece tokenizer? Word
+    piece tokenizer? Is it the same tokenizer as used for BERT or BART?
+
+After you feel like you have gotten a good overview of the architecture
+of the model, you might want to write to Suraj with any
+questions you might have. This might include questions regarding the
+model's architecture, its attention layer, etc. We will be more than
+happy to help you.
+
+
+#### Additional resources
+
+ Before diving into the code, here are some additional resources that might be worth taking a look at:
+ 
+ - [link 1]
```

Comment:
those should be filled out with some useful information on what to read up on for GLM